fix: black screen in ExoPlayer on rotate

k1rill opened the pull request:
Workaround with VideoPlayer flow on configuration changes.

What was done:
1. Shared EncodedVideoUnitViewModel between VideoUnitFragment and VideoFullScreenFragment and storing exoPlayer instance in it.
2. VideoFullScreenFragment became DialogFragment to have access to the shared viewModel
3. VideoFullScreenFragment moved to Compose
4. State in flow in  EncodedVideoUnitViewModel

k1rill committed:
fix: black screen on exoplayer on rotate

HamzaIsrar12 commented:
I conducted a manual review on the emulator last week, where things still felt quite laggy. However, I tested it again on my own device today, and the experience was much improved. The lag decreased from 5–6 seconds to 1–2 seconds. Great job on that!

I did notice two issues from a learner's perspective:  
- Rotating the phone to change orientation in fullscreen causes the video to pause, which shouldn’t happen.  
- The on-player captions aren’t working at all. (For clarity, I’ll refer to the on-player captions as subtitles and the captions below the player as transcription.)

_Note: @omerhabib26  is performing an in-depth review._

omerhabib26 commented:
Hey @k1rill, great job on the revamp! The lag has dropped to just 1-2 seconds, which is a big improvement. However, there are still a couple of issues that need attention:

- When the device is in portrait mode and the user taps on the fullscreen button, the player switches to fullscreen, but the video size doesn’t adjust—it stays in portrait orientation. Ideally, the player should expand to full-screen size and switch to landscape mode. A good example of this behavior is the YouTube app.

- When the device is in portrait mode and the user rotates it, a white gap appears at the bottom of the screen, covering the player’s bottom options.


https://github.com/user-attachments/assets/85c1bb08-e0f0-4499-9ccb-83ff66c9cd82

k1rill committed:
fix: exoplayer in view model speed up

k1rill committed:
fix: subtitles in fullscreen player

k1rill commented:
@HamzaIsrar12 @omerhabib26 I just updated my PR with few commit. 

- I increased speed of the player
- I fixed mentioned issue, also subtitles are working now

Ready for pass

omerhabib26 commented:
Reviewing.

k1rill committed:
fix: pause player on rotation

omerhabib26 commented on course/src/main/java/org/openedx/course/presentation/unit/video/VideoUnitFragment.kt:
```diff
@@ -225,19 +208,12 @@ class VideoUnitFragment : Fragment(R.layout.fragment_video_unit) {
-                )
+                viewModel.isPlaying = viewModel.getActivePlayer()?.isPlaying.isTrue()
+                viewModel.enterFullscreen()
+                VideoFullScreenFragment.newInstance().show(childFragmentManager, VideoFullScreenFragment.TAG)
```
IMO, better to move this to viewModel

omerhabib26 reviewed (changes requested):
Hey @k1rill,
Thanks for the update and great work on this issue. As per the current implementation, the app is working smoothly for native videos, but for YouTube videos, the app still has some lags and is causing the same issue we previously had with native videos.
Otherwise, only some minor nits are needed.

miankhalid commented:
@omerhabib26 isn't YT player out-of-scope for this PR? or am I missing something here?

or are you referring to something that has changed in the PR that is now causing lags for the YT video player?

omerhabib26 commented:
The issue with the video component was a time lag on rotation, regardless of whether it was a native or YouTube player. Since both use the same approach and view models, that's why I'm asking about the behavior for YouTube as well.

k1rill committed:
chore: changed enterFullscreen method behaviour
